src/zenml/utils/json_utils.py

```diff
@@ -0,0 +1,125 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Carried over version of some functions from the pydantic v1 json module.
+
+Check out the latest version here:
+https://github.com/pydantic/pydantic/blob/v1.10.15/pydantic/json.py
+"""
+
+import datetime
+from collections import deque
+from decimal import Decimal
+from enum import Enum
+from ipaddress import (
+    IPv4Address,
+    IPv4Interface,
+    IPv4Network,
+    IPv6Address,
+    IPv6Interface,
+    IPv6Network,
+)
+from pathlib import Path
+from re import Pattern
+from types import GeneratorType
+from typing import Any, Callable, Dict, Type, Union
+from uuid import UUID
+
+from pydantic import NameEmail, SecretBytes, SecretStr
+from pydantic.color import Color
+
+__all__ = "pydantic_encoder"
+
+
+def isoformat(o: Union[datetime.date, datetime.time]) -> str:
+    """Function to convert a datetime into iso format.
+
+    Args:
+        o: input datetime
```

Comment:
why `o`? Generally I dislike single letter variables, but o specifically is just random :laughing:

Reply:
Carried the code over from pydantic v1, along with the parameter names 😄